org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/formatter/FormatterTestHelper.java

```diff
@@ -138,18 +139,45 @@ protected void assertNoSyntaxErrors(XtextResource resource) {
 	protected void assertReplacementsAreInRegion(List<ITextReplacement> rep, Collection<ITextRegion> regions,
 			String doc) {
 		Set<ITextReplacement> invalid = Sets.newHashSet();
-		ALLOWED: for (ITextRegion allowed : regions)
-			for (ITextReplacement r : rep) {
-				if (allowed.contains(r))
+		ALLOWED: for (ITextRegion region : regions) {
+			Iterator<ITextReplacement> iterator = rep.iterator();
+			while (iterator.hasNext()) {
+				ITextReplacement replacement = iterator.next();
+				if (region.contains(replacement) || doIntersect(region, replacement))
 					continue ALLOWED;
-				invalid.add(r);
+				
+				if(!iterator.hasNext())
+					invalid.add(replacement);
 			}
+		}
+		
 		if (!invalid.isEmpty()) {
 			String visualized = new TextRegionsToString().addAllReplacements(invalid).toString();
 			fail("One or more TextReplacements are outside of the allowed region. Region: " + regions, visualized);
 		}
 	}
 
+	private boolean doIntersect(ITextRegion allowed, ITextReplacement replacement) {
+		int allowedOffset = allowed.getOffset();
+		int replacementOffset = replacement.getEndOffset();
+
+		if(allowedOffset == replacementOffset) {
+			return true;
```

Comment:
*touching* regions don't overlap, do they? I think this case shouldn't be considered true.